rustls-bench/src/main.rs

```diff
@@ -785,6 +756,66 @@ fn lookup_matching_benches(name: &str) -> Vec<&BenchmarkParam> {
     r
 }
 
+/// General parameters common to several kinds of
+/// benchmark.
+#[derive(Clone)]
+struct Parameters {
+    /// Set by the user.
+    options: Options,
```

Comment:
Not sure if the `Options` type still makes sense? I thought that was directly used for argument parsing, but I noticed just now it is instantiated separately. I'd review whether it makes more sense to flatten it into `Parameters`.

Or, maybe these can actually be fields on `Parameters` instead of being recreated?

Reply:
Removed `Options` and inlined into `Parameters`. That means we need to resolve the provider in several places, but I think that's OK.